pkg/azuredisk/controllerserver_v2.go

```diff
@@ -488,7 +532,9 @@ func (d *DriverV2) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest)
 			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) can not be negative", start)
 		}
 	}
-	if d.cloud.KubeClient != nil && d.cloud.KubeClient.CoreV1() != nil && d.cloud.KubeClient.CoreV1().PersistentVolumes() != nil {
+
+	kubeClient := d.controllerProvisioner.GetKubeClient()
```

Comment:
Since this largely deals with listing PVs contained in Azure-specific resources, I expect most of the implementation could move to the provisioner.